<launch>

  <arg name="uav_name" default="$(optenv UAV_NAME uav1)"/>
  <arg name="uav_id" default="$(optenv UAV_ID 0)"/>
    
    <!-- will it run using GNU debugger? -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg debug)" name="launch_prefix_debug" value="debug_roslaunch"/>

  <group ns="$(arg uav_name)">

    <!-- <node name="newUvdarRx" pkg="uvdar_core" type="new_uvdar_rx_node" output="screen" launch-prefix="$(arg launch_prefix_debug)" > -->
    <node name="newUvdarRx" pkg="uvdar_core" type="new_uvdar_rx_node" output="screen" launch-prefix="debug_roslaunch" >

      <param name="uav_name" value="$(arg uav_name)" />
      <param name="uav_id" value="$(arg uav_id)" />
      <param name="recieved_topic" value="/$(arg uav_name)/uvdar_communication/recieved_msgs" />
      <rosparam param="points_seen_topics"> ["points_seen_left", "points_seen_right"] </rosparam>
      <rosparam param="blinkers_seen_topics"> ["blinkers_seen_left", "blinkers_seen_right"] </rosparam>
      <rosparam param="estimated_framerate_topics"> ["estimated_framerate_left", "estimated_framerate_right"] </rosparam>
      <remap from="~points_seen_left" to="/$(arg uav_name)/uvdar/points_seen_left"/>
      <remap from="~points_seen_right" to="/$(arg uav_name)/uvdar/points_seen_right"/>
      <remap from="~blinkers_seen_left" to="/$(arg uav_name)/uvdar/blinkers_seen_left"/>
      <remap from="~blinkers_seen_right" to="/$(arg uav_name)/uvdar/blinkers_seen_right"/>
      <remap from="~estimated_framerate_left" to="/$(arg uav_name)/uvdar/estimated_framerate_left"/>
      <remap from="~estimated_framerate_right" to="/$(arg uav_name)/uvdar/estimated_framerate_right"/>
    </node>
    
  </group>
  
</launch>
